feat(curriculum): add fortune teller lab (#55888)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/lab-fortune-teller/66c06d618d075c7f7f1b890a.md

---
id: 66c06d618d075c7f7f1b890a
title: Build a Fortune Teller
challengeType: 14
dashedName: build-a-fortune-teller
---

# --description--

In this lab, you will define five fortunes and randomly select one of them to display the fortune to the user.

> **Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. 

Reviewer: ```suggestion
**Objective:** Complete the following user stories and ensure all tests pass to complete the lab.
```

Reviewer 2: We are using that sentence for all the labs @Sembauke Personally, I'd keep it as is.

Reviewer 3: I would just keep it like that then too. We should use that for now on all the labs and we can go back and change them all at once if we want some else.

---
id: 66c06d618d075c7f7f1b890a
title: Build a Fortune Teller App
challengeType: 14
dashedName: build-a-fortune-teller-app
---

# --description--

**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. 

**User Stories:**

1. You should initialize the five variables `fortune1`, `fortune2`, `fortune3`, `fortune4`, and `fortune5` with a string value of your choice. You can use the below options if you like:
    

> - `"Your cat will look very cuddly today."`.
> - `"The weather will be nice tomorrow."`.
> - `"Be cautious of your new neighbours."`.
> - `"You will find a new hobby soon."`.
> - `"It would be wise to avoid the colour red today."`.

Reviewer 3: ```suggestion
- `"Your cat will look very cuddly today."`
- `"The weather will be nice tomorrow."`
- `"Be cautious of your new neighbours."`
- `"You will find a new hobby soon."`
- `"It would be wise to avoid the colour red today."`
```

Pretty minor, but two periods in a row looks weird to me.

> 1. You should select a random number between 1 and 5, inclusive, and assign it to the variable `randomNumber`.

Reviewer 3: ```suggestion
2. You should select a random number between 1 and 5, inclusive, and assign it to the variable `randomNumber`.
```

When you put a list in between your numbers, it thinks you're starting a new list of numbers - so you have to put the specific numbers in this case.

Author: I am getting a linting error after updating these.

> You should use an `if/else if` statement to print a different fortune based on the value of `randomNumber`:

Reviewer 3: I would remove this user story and the tests - and let users print the variable in any way they like.

Author: Done.

> - If `randomNumber` is 1, print the first fortune.
> - If `randomNumber` is 2, print the second fortune.
> - If `randomNumber` is 3, print the third fortune.
> - If `randomNumber` is 4, print the fourth fortune.
> - If `randomNumber` is 5, print the fifth fortune.
> 
> 1. You should log the fortune to the console.

Reviewer 3: ```suggestion
3. You should log the fortune to the console in this manner:

- If `randomNumber` is 1, print the `fortune1`.
- If `randomNumber` is 2, print the `fortune2`.
- If `randomNumber` is 3, print the `fortune3`.
- If `randomNumber` is 4, print the `fortune4`.
- If `randomNumber` is 5, print the `fortune5`.
```

I would combine these somehow.

Author: Reworded these statements.

   
# --hints--

You should initialize `fortune1` with a string value.

```js
assert.isNotNull(fortune1);
assert.isString(fortune1);
```

You should initialize `fortune2` with the a string value.

```js
assert.isNotNull(fortune2);
assert.isString(fortune2);
```

You should initialize `fortune3` with a string value.

```js
assert.isNotNull(fortune3);
assert.isString(fortune3);
```

You should initialize `fortune4` with a string value.

```js
assert.isNotNull(fortune4);
assert.isString(fortune4);
```

You should initialize `fortune5` with a string value.

```js
assert.isNotNull(fortune5);
assert.isString(fortune5);
```

You should generate a random number between 1 and 5, inclusive, and assign it to the variable `randomNumber`.

```js
assert.isAtLeast(randomNumber, 1);

> assert.isAtMost(randomNumber, 5);

Reviewer 3: I would include a regex test for `math.random` or something - so they don't just assign it a number.

Author: Regex has been included.

```js
assert.match(__helpers.removeJSComments(code), /if\s*\(randomNumber\s*===\s*1\)\s*{\s*selectedFortune\s*=\s*fortune1;\s*}/);
```

You should create another condition using `else if` to check if `randomNumber` is equal to `2` and assign `selectedFortune` to `fortune2`.

```js
assert.match(__helpers.removeJSComments(code), /else\s*if\s*\(randomNumber\s*===\s*2\)\s*{\s*selectedFortune\s*=\s*fortune2;\s*}/);
```

You should create another condition using `else if` to check if `randomNumber` is equal to `3` and assign `selectedFortune` to `fortune3`.

```js
assert.match(__helpers.removeJSComments(code), /else\s*if\s*\(randomNumber\s*===\s*3\)\s*{\s*selectedFortune\s*=\s*fortune3;\s*}/);
```

You should create another condition using `else if` to check if `randomNumber` is equal to `4` and assign `selectedFortune` to `fortune4`.

```js
assert.match(__helpers.removeJSComments(code), /else\s*if\s*\(randomNumber\s*===\s*4\)\s*{\s*selectedFortune\s*=\s*fortune4;\s*}/);
```

You should create a final condition using `else` to assign `selectedFortune` to `fortune5`.

```js
assert.match(__helpers.removeJSComments(code), /else\s*{\s*selectedFortune\s*=\s*fortune5;\s*}/);

> ```

Reviewer 3: If you remove the if/else user story, you can remove these tests.

Author: I have removed the `if/esle` story and these tests.

You should output the `selectedFortune` to the console.

```js

> assert.match(__helpers.removeJSComments(code), /console\.log\(selectedFortune\);?/);

Reviewer 3: ```suggestion
assert.match(__helpers.removeJSComments(code), /console\s*\.\s*log\s*\(\s*selectedFortune\s*\)\s*;?/);
```

This makes it a little more forgiving by allowing spaces where JS allows.
